### PR TITLE
Move remaining System.ComponentModel.Annotations tests to test base

### PR DESCRIPTION
**Pull request by hughbe**
/cc @stephentoub

### CODE REVIEW

File: src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs [hunk 1]
```diff
@@ -29,181 +88,53 @@ public static void Can_construct_and_get_minimum_and_maximum_for_double_construc
...
-            Assert.Throws<ValidationException>(() => attribute.Validate((3.0000001).ToString(), s_testValidationContext));
+            var attribute = new RangeAttribute(type, "someMinimum", "someMaximum");
+            Assert.Throws<InvalidOperationException>(() => attribute.Validate("Any", new ValidationContext(new object())));
         }
 
         [Fact]
-        public static void Validation_throws_InvalidOperationException_for_null_OperandType()
-        {
-            var attribute = new RangeAttribute(null, "someMinimum", "someMaximum");
-            Assert.Null(attribute.OperandType);
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs]: Where did this go?

File: src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs [hunk 2]
```diff
@@ -29,181 +88,53 @@ public static void Can_construct_and_get_minimum_and_maximum_for_double_construc
...
-            Assert.Throws<InvalidOperationException>(
-                () => attribute.Validate("Does not matter - OperandType is null", s_testValidationContext));
-        }
-
-        [Fact]
-        public static void Validation_throws_InvalidOperationException_for_OperandType_which_is_not_assignable_from_IComparable()
-        {
-            var attribute = new RangeAttribute(typeof(InvalidOperandType), "someMinimum", "someMaximum");
-            Assert.Equal(typeof(InvalidOperandType), attribute.OperandType);
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs]: Where did this go?

File: src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs [hunk 3]
```diff
@@ -2,78 +2,57 @@
...
-        public static void Validate_throws_for_values_which_do_not_match_pattern()
-        {
-            var attribute = new RegularExpressionAttribute("defghi");
-            Assert.Throws<ValidationException>(() => attribute.Validate("zyxwvu", s_testValidationContext)); // pattern does not match
-            Assert.Throws<ValidationException>(() => attribute.Validate("defghijkl", s_testValidationContext)); // pattern only matches beginning of value
-            Assert.Throws<ValidationException>(() => attribute.Validate("abcdefghi", s_testValidationContext)); // pattern only matches end of value
-            Assert.Throws<ValidationException>(() => attribute.Validate("abcdefghijkl", s_testValidationContext)); // pattern only matches part of value
-
-            attribute = new RegularExpressionAttribute("[^a]+\\.[^z]+");
-            attribute.MatchTimeoutInMilliseconds = 10000; // note: timeout is just a number much larger than we expect the test to take
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs]: Are there any tests for when the timeout is really low, like 0, and the regex evaluation times out?

File: src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs]: here: https://github.com/hughbe/corefx/blob/0f51ae6792f535f9fe0dbb724c33c003ac657910/src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs#L96

File: src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs]: will add back.

**Comment by stephentoub** [comment]
Thanks.  A few comments, otherwise LGTM.

File: src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RangeAttributeTests.cs]: added back

File: src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs (hunk 3 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs]: I added a test for a timeout of zero (throws ArgumentOutOfRangeException, which is probably a bug!)
However, I can't figure out how to make the regex evaluation to timeout - even something like this; 
```

yield return new TestCase(new RegularExpressionAttribute(new string('a', 50000)) { MatchTimeoutInMilliseconds = 1 }, new string('a', 50000));
```
passes!

Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs]: That's a very fast pattern to match; you need a more complicated pattern that takes a long time to evaluate, likely one involving backtracking.  Try this, for example:
```C#
Regex.IsMatch("aaaaaaaaaaaaaaaaaaaaaaaaaaaa>", "(a+)+$");
```
and watch your CPU spike.

Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs]: cheers

Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs]: i've added this test, cheera

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**